docs/tutorials/gluon/logistic_regression_explained.md

```diff
@@ -0,0 +1,215 @@
+    
+    Epoch: 6, Training loss: 54.76, Validation loss: 7.45, Validation accuracy: 0.98 <!--notebook-skip-line-->
+    
+    Epoch: 7, Training loss: 48.29, Validation loss: 8.56, Validation accuracy: 0.97 <!--notebook-skip-line-->
+    
+    Epoch: 8, Training loss: 50.50, Validation loss: 7.24, Validation accuracy: 0.98 <!--notebook-skip-line-->
+    
+    Epoch: 9, Training loss: 49.42, Validation loss: 7.46, Validation accuracy: 0.97 <!--notebook-skip-line-->
+
+
+## Tip 1: Use only one neuron in the output layer
+
+Despite that there are 2 classes, there should be only one output neuron, because `SigmoidBinaryCrossEntropyLoss` accepts only one feature as an input. 
+
+In case when there are 3 or more classes, one cannot use a single Logistic regression, but should do multiclass regression. The solution would be to increase the number of output neurons to the number of classes and use `SoftmaxCrossEntropyLoss`. 
+
+## Tip 2: Encode classes as 0 and 1
+
+`Sigmoid` function produces values from 0 to 1. `SigmoidBinaryCrossEntropyLoss` uses these values to calculate the   loss by essentially subtracting values and class labels from 1. [Here is the formula](https://mxnet.incubator.apache.org/api/python/gluon/loss.html?highlight=sigmoidbinarycrossentropyloss#mxnet.gluon.loss.SigmoidBinaryCrossEntropyLoss) used for that calculation (we use default version with `from_sigmoid` is False). That's why it is numerically better to have classes encoded in the same range as a `Sigmoid` output with 0 and 1.
+
+If your data comes with a label encoded in a different format, such as -1 and 1, then you can either recode it to 0 and 1 by comparing the initial class to 0, or use another function instead of `Sigmoid`, like [`Tanh`](https://mxnet.incubator.apache.org/api/python/ndarray/ndarray.html?highlight=tanh#mxnet.ndarray.tanh), to produce output in range [-1; 1].
+
+## Tip 3: Use SigmoidBinaryCrossEntropyLoss instead of LogisticRegressionOutput
+
+NDArray API has two options to calculate logistic regression loss. One is `SigmoidBinaryCrossEntropyLoss`, which I used in the example. This class inherits from the `Loss` class and is intended to be used as a loss function for logistic regression. But there is also a function called `LogisticRegressionOutput`, which can be applied to any `NDArray`. Mathematically speaking, this function does the same thing as `SigmoidBinaryCrossEntropyLoss`. 
+
+My recommendation would be to use `SigmoidBinaryCrossEntropyLoss`, because this class properly inherits from `Loss` class, while `LogisticRegressionOutput` is just a regular function. `LogisticRegressionOutput` is a   function to go when implementing logistic regression using Symbol API, but in case of using Gluon API, there are no benefits using it. The only case when you may want to consider using `LogisticRegressionOutput` as your loss, is when you need to have a support for sparse matrices.
```

Comment:
This paragraph needs rewording. It doesn't really make a case for why not use LogisticRegressionOutput (i.e. the fact that it is designed to be an output layer when using the Module API). Also what's the problem with using sparse matrices and SigmoidBinaryCrossEntropyLoss?! What happens if I do? What fails? You may consider just getting rid of this tip altogether to avoid any confusion.

Reply:
If you try to use `SigmoidBinaryCrossEntropyLoss` with sparse data, it would throw an exception at some point, where it tries to do a reshape:

```NotSupportedForSparseNDArray: Function reshape with arguments (<class 'tuple'>) is not supported for SparseNDArray and only available in NDArray.```

Reply:
I rewrote the paragraph, and I actually deleted the sparse part from it as well. @eric-haibin-lin is going to work on this at some point, so to not update the tutorial later, I just omit this part.